### ISSUE TITLE
Monthly Meetup - April 2019

### ISSUE THREAD

**Issue by chihab** [rabatjs/meetups#2]
### Cracking a JavaScript Interview - Beginner Series by @chihab [ Help wanted ]
You're already a JavaScript developer but do you really feel comfortable with things like scope, closures, this, call, apply, bind, arrow functions, event loop, async await...? Are you able to explain these concepts if you're asked to in an interview?
In this Series we'll get onto the concepts that every JavaScript Developer should know. No slides, just an editor.

### Functional Reactive Programming with RxJS - Intermediate by @chihab 
In this talk we will get introduced to RxJS and the Observable type. An Observable is a 'collection/array that arrives over time'. It can be used to model events, async requests.... Observables can be transformed, combined, and consumed using operators (functions that look like JavaScript Array.prototype methods).

## Event
https://www.meetup.com/Rabat-js/events/260532405/

## Agenda
6:15 pm - Doors open
6:30 pm - Talks begin!
8:00 pm - Discussions / Neworking
8:30 pm - See you next time!

## When/Where
April 17th at KidLabs - Rabat

**Comment by chihab** [comment]
I have been asked (by fellow members ;) to do beginner topics along with intermediate / advanced ones. What I suggest is to have two talks on each meetup.

* 60mn for a beginner topic
   Example: Mini workshops, Intro to JavaScript, React components and props, Intro to GraphQL, Intro to TypeScript, Simple Rest API with express...
* 30mn for a intermediate / advanced topic
   Example: Functional programming (composition, currying, pure func...), RxJS workshop, Redux / Mobx / Apollo Cache, Advanced Component patterns in React / Angular, Unit Tests, Webpack....

The two talks might or might not be related.

**Comment by akiyamaSM** [comment]
Only React related?

**Comment by chihab** [comment]
@akiyamaSM Not at all ! All JS related stuff be it the language itself, JS libraries/frameworks like Angular, React, Vue, JS plateforms like Browser, Node and Mobile... cloud stuff like lambdas and so...as long as it is interesting for JS developers so no Java/PHP/C#...

See #1.

**Comment by akiyamaSM** [comment]
@chihab Thanks. So its good as long as its JS related. Actually I have some experience with vuejs. And I don't know if I may be of any help.

**Comment by chihab** [comment]
@akiyamaSM of course. If you'd like to give a talk about some vuejs topic propose it right here in a comment. Give it a name, a description and how much time you'd need (could be from 10mn to 60mn).

**Comment by akiyamaSM** [comment]
Thanks, If there is any news I'll keep you updated in here.
